src/compiler/transformers/ts.ts

```diff
@@ -2817,32 +2820,28 @@ namespace ts {
                 //     module m1 {
                 //         function foo4Export() {
                 //         }
-                //     } // trailing comment module
+                //     } // trailing module comment
                 //
                 // Should emit:
                 //
                 //     /** Module comment*/
-                //     var m1;
+                //     var m1 = m1 || {};
                 //     (function (m1) {
                 //         function foo4Export() {
                 //         }
-                //     })(m1 || (m1 = {})); // trailing comment module
+                //     })(m1); // trailing module comment
                 //
                 setCommentRange(statement, node);
-                setEmitFlags(statement, EmitFlags.NoTrailingComments | EmitFlags.HasEndOfDeclarationMarker);
+                setEmitFlags(statement, EmitFlags.NoTrailingComments);
                 statements.push(statement);
+                if (nodeIsExportDefault) {
+                    statements.push(createExportAssignment(/* decorators */ undefined, /* modifiers */ undefined, /* isExportEquals */ false, localName));
+                }
+                else if (!currentNamespace && nodeModifierFlags & ModifierFlags.Export) {
+                    statements.push(createExternalModuleExport(localName));
+                }
                 return true;
             }
-            else {
-                // For an EnumDeclaration or ModuleDeclaration that merges with a preceeding
-                // declaration we do not emit a leading variable declaration. To preserve the
-                // begin/end semantics of the declararation and to properly handle exports
-                // we wrap the leading variable declaration in a `MergeDeclarationMarker`.
-                const mergeMarker = createMergeDeclarationMarker(statement);

```

Comment:
MergeDeclarationMarker are unnecessary if the variable is initialized immediately. I think the logic handling those in the other modules is now dead.